src/note_compiler.c

```diff
@@ -226,3 +229,43 @@ int NoteCompiler_get_next_note(void* compiler, Note* note, bool* finished) {
 	return 0;
 }
 
+int NoteCompiler_print_error(const NoteCompiler* compiler, FILE* stream) {
+
+	int status = 0;
+
+	if(compiler == NULL || !(compiler->error)) {
+		return ERROR_CODE_INVALID_ARGUMENT;
+	}
+
+	if(compiler->bar == NULL || compiler->bar->content == NULL) {
+		return ERROR_CODE_INVALID_STATE;
+	}
+
+	fprintf(stream, "%s:%u:%u: ", compiler->filename, compiler->bar->line, compiler->bar->col);
+
+	switch(compiler->error_state) {
+
+		case NOTE_COMPILER_ERROR_STATE_UNEXPECTED_CHARACTER:
+
+			fputs("Unexpected character:\n", stream);
+			BarToken_print(compiler->bar, stream);
+
+			if(fprintf(stream, "\n%*s^", (int) (compiler->position - 1), " ") < 0) {
```

Comment:
We're likely gonna use the exact same trick inside the `token` module; might be worth extracting the string formatting into a helper method.